ProcessLib/SmallDeformation/SmallDeformationFEM.h

```diff
@@ -384,49 +387,19 @@ class SmallDeformationLocalAssembler
 
     std::size_t setSigma(double const* values)
     {
-        auto const kelvin_vector_size =
-            MathLib::KelvinVector::KelvinVectorDimensions<
-                DisplacementDim>::value;
-        auto const n_integration_points = _ip_data.size();
-
-        auto sigma_values =
-            Eigen::Map<Eigen::Matrix<double, kelvin_vector_size, Eigen::Dynamic,
-                                     Eigen::ColMajor> const>(
-                values, kelvin_vector_size, n_integration_points);
-
-        for (unsigned ip = 0; ip < n_integration_points; ++ip)
-        {
-            _ip_data[ip].sigma =
-                MathLib::KelvinVector::symmetricTensorToKelvinVector(
-                    sigma_values.col(ip));
-        }
-
-        return n_integration_points;
+        return ProcessLib::setKelvinVectorValues<
+            IpData, decltype(&IpData::sigma), DisplacementDim>(values, _ip_data,
+                                                               &IpData::sigma);
     }
 
     // TODO (naumov) This method is same as getIntPtSigma but for arguments and
     // the ordering of the cache_mat.
     // There should be only one.
     std::vector<double> getSigma() const override
     {
-        auto const kelvin_vector_size =
-            MathLib::KelvinVector::KelvinVectorDimensions<
-                DisplacementDim>::value;
-        auto const n_integration_points = _ip_data.size();
-
-        std::vector<double> ip_sigma_values;
-        auto cache_mat = MathLib::createZeroedMatrix<Eigen::Matrix<
-            double, Eigen::Dynamic, kelvin_vector_size, Eigen::RowMajor>>(
-            ip_sigma_values, n_integration_points, kelvin_vector_size);
-
-        for (unsigned ip = 0; ip < n_integration_points; ++ip)
-        {
-            auto const& sigma = _ip_data[ip].sigma;
-            cache_mat.row(ip) =
-                MathLib::KelvinVector::kelvinVectorToSymmetricTensor(sigma);
-        }
-
-        return ip_sigma_values;
+        return ProcessLib::getKelvinVectorValues<
+            IpData, decltype(&IpData::sigma), DisplacementDim>(_ip_data,
+                                                               &IpData::sigma);
```

Comment:
I think, if you reorder the template arguments of `getKelvinVectorValues` to`
```c++
template <int DisplacementDim, typename IpData, typename Member>
std::vector<double> getKelvinVectorValues(IpData const&, Member const&);
```
then you can omit the `IpData` and the member pointer types.

Reply:
Going to try your suggestion.

Reply:
It works.